src/AdvancedPureComponent.js

```diff
@@ -10,7 +10,7 @@ const createAdvancedPureComponent = (Component, toFlattenProps = _ => _) => {
     render() {
       const { children, ...props } = this.props
       return (
-        <Component title={children} {...props}>
+        <Component title={props.invalid || children} {...props}>
```

Comment:
I am confused as to what this  is suppose to do. it could be one of 2 things
1. you want the title to show a invalid message instead of the children. by passing in `invalid='something is wrong'`
2. you don't want the title to show by passing in `invalid=true`

In order to make this clear let's do the following
If you mean to implement 1 then use. `titleMessage`
if you mean to implement 2 then use `titleDisabled`

Reply:
ok, i'm just changing it to invalidMessage, so invalid should return invalid Message or false in case of valid.